yoyodyne/predict.py

```diff
@@ -1,50 +1,142 @@
 """Prediction."""
 
+import argparse
 import os
-from typing import Optional
 
-import click
 import pytorch_lightning as pl
 import torch
 from torch.utils import data
 
 from . import collators, dataconfig, datasets, models, util
 
 
-def write_predictions(
-    model: models.base.BaseEncoderDecoder,
-    loader: torch.utils.data.DataLoader,
-    output: str,
-    accelerator: str,
+def get_trainer(**kwargs) -> pl.Trainer:
+    """Creates the trainer.
+
+    Args:
+        **kwargs: passed to the trainer.
+
+    Returns:
+        pl.Trainer.
+    """
+    return pl.Trainer(max_epochs=0, **kwargs)
+
+
+def _get_trainer_from_argparse_args(
+    args: argparse.Namespace,
+) -> pl.Trainer:
+    """Creates the trainer from CLI arguments."""
+    return pl.Trainer.from_argparse_args(args, max_epochs=0)
+
+
+def get_dataset(
+    predict: str,
+    config: dataconfig.DataConfig,
+    index: str,
+) -> data.Dataset:
+    """Creates the dataset.
+
+    Args:
+        predict (str).
+        config (dataconfig.DataConfig).
+        index (str).
+
+    Returns:
+        data.Dataset.
+    """
+    # TODO: Since we don't care about the target column, we should be able to
+    # set config.source_col = 0 and avoid the overhead for parsing it.
+    # This does not work because the modules expect it to be present even if
+    # they ignore it.
+    dataset = datasets.get_dataset(predict, config)
```

Comment:
This still loads an initial index since we don't have a different dataset to pass, right? I think if we could pass the index (in this case the index file it seems) to the Dataset this could be avoided (future PR?)

Reply:
Oh great idea. I'll add a TODO.